src/test/java/com/cgm/infolab/WebSocketDeleteEditPublicTests.java

```diff
@@ -0,0 +1,343 @@
+                    WebSocketMessageDto received = receivedMessages.poll();
+
+                    Assertions.assertNotNull(received.getEdit());
+                    Assertions.assertEquals(webSocketEditedMessage.getEdit().getId(), received.getEdit().getId());
+                    Assertions.assertEquals(EDIT, received.getType());
+
+                    Assertions.assertNull(received.getChat());
+                    Assertions.assertNull(received.getDelete());
+
+                    ChatMessageEntity messageEntity = testDbHelper
+                            .getAllMessages()
+                            .stream()
+                            .filter(message -> message.getId() == messageId)
+                            .toList()
+                            .get(0);
+
+                    Assertions.assertEquals(EDITED, messageEntity.getStatus());
+                    Assertions.assertEquals("New content : )", messageEntity.getContent());
+                });
+    }
+
+    @Test
+    void whenEditFails_websocketReturnsNull() throws ExecutionException, InterruptedException, TimeoutException {
+        StompSession client = getStompSession();
+
+        BlockingQueue<WebSocketMessageDto> receivedMessages = new ArrayBlockingQueue<>(2);
+
+        client.subscribe("/topic/public", new StompFrameHandler() {
+            @Override
+            public Type getPayloadType(StompHeaders headers) {
+                return WebSocketMessageDto.class;
+            }
+
+            @Override
+            public void handleFrame(StompHeaders headers, Object payload) {
+                receivedMessages.add((WebSocketMessageDto) payload);
+                client.disconnect();
+            }
+        });
+
+        long messageId = 4;
+        ChatMessageEntity messageEntityBefore = testDbHelper
+                .getAllMessages()
+                .stream()
+                .filter(message -> message.getId() == messageId)
+                .toList()
+                .get(0);
+
+        WebSocketMessageDto webSocketEditedMessage = WebSocketMessageDto.ofEdit(ChatMessageDto.of(messageId, "Edit that fails"));
+        client.send("/app/chat.edit", webSocketEditedMessage);
+
+        await()
+                .atMost(1, TimeUnit.SECONDS)
+                .untilAsserted(() -> {
+                    ChatMessageEntity messageEntityAfter = testDbHelper
+                            .getAllMessages()
+                            .stream()
+                            .filter(message -> message.getId() == messageId)
+                            .toList()
+                            .get(0);
+
+                    Assertions.assertEquals(messageEntityBefore, messageEntityAfter);
+
+                    WebSocketMessageDto received = receivedMessages.poll();
+
+                    Assertions.assertNull(received);
+                });
+    }
+
+    private StompSession getStompSession() throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
Se questo metodo privato è uguardo a quello di `InfolabApplicationTests.java` estrailo in un helper